gxutil/lckfile.go

```diff
@@ -0,0 +1,28 @@
+package gxutil
+
+import (
+	"encoding/json"
+	"io/ioutil"
+)
+
+type LockDep struct {
+	// full ipfs path /ipfs/<hash>/foo
+	Ref string
+
+	// Mapping of dvcs import paths to LockData
+	Deps map[string]LockDep
+}
+
+type LockFile struct {
+	Language string
+	Deps     map[string]LockDep
+}
+
```

Comment:
Per our discussion:

```go
type Lock struct {
    // Empty for the top level package.
    // In the future, we could use IPNS!
    Ref string
    Language string

    // maps language namespace -> name in language -> dependency
    Deps map[string]map[string]Lock
}

Reply:
(or we could keep it non-recursive, that was just a thought)

Reply:
Why the "double map": The initial proposal was `"dep name in language" -> definition`. For example:

```
{
    "github.com/ipfs/go-ipfs": {
        "ref": "/ipfs/....",
        "language": "go"
    }
}
```

Unfortunately, this falls apart when we need multiple namespaces. For example, with CGO, we might want (not well thought out):

```json
{
    "github.com/ipfs/go-ipfs": {
        "ref": "/ipfs/....",
        "language": "go"
    },
    "libs/libsnark.a": {
        "ref": "/ipfs/....",
        "language": "rust"
    }
}
```

```go
package thing

// #cgo LDFLAGS: -L${SRCDIR}/libs -llibsnark
import "C"
```

The problem here is that there's the "go package" namespace and the "cgo" namespace (really, just the filesystem) and we need some way to disambiguate.

Reply:
The double map would be:

```
{
    "go": {
        "github.com/ipfs/go-ipfs": {
            "ref": "/ipfs/....",
            "language": "go"
        }
    },
    "cgo": { // files?
        "libs/libsnark.a": {
            "ref": "/ipfs/....",
            "language": "rust"
        }
    }
}
```